### DISCUSSION TITLE
Gems - Path to /bin?

### DISCUSSION THREAD

**Discussion by patlux** [jdx/mise#617] [Q&A]
How do you manage to add the path of your gems into your $PATH?

I just managed to do it like this:

```sh
# .zshrc

eval \"$(~/.bin/rtx activate zsh)\"
export GEM_HOME=$(rtx exec ruby --command 'ruby -e \"puts Gem.user_dir\"')
export PATH=$PATH:$GEM_HOME/bin 
```

is this the correct way?

**Comment by Marukome0743** [comment]
With a Ruby version managed by mise, an additional `GEM_HOME` normally should not be necessary.

A regular:

```sh
gem install <gem>
```

installs the gem into the currently selected Ruby installation. Its executable directory is already covered by the Ruby `bin` directory that mise places on `PATH`.

Setting `GEM_HOME` to `Gem.user_dir` in `.zshrc` can be problematic because it is calculated only when the shell starts. After changing into a project that selects a different Ruby version, it may still point at the directory calculated for the previous version.

The simplest setup is therefore:

```sh
eval "$(mise activate zsh)"
mise use ruby@3.4
gem install <gem>
```

If the gem is primarily a standalone CLI and should be managed independently of a particular Ruby installation, current mise also provides the `gem:` backend:

```toml
[tools]
"gem:rubocop" = "latest"
```

In that case, mise exposes the installed executable directly and handles its runtime dependency.

https://mise.en.dev/dev-tools/backends/gem.html

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
